// @mysten/wallet-adapter-sui-wallet/dist/index.mjs
// src/adapter.ts
var SuiWalletAdapter = class {
  constructor() {
    this.name = "Sui Wallet (legacy)";
    this.connected = false;
    this.connecting = false;
  }
  getAccounts() {
    return window.suiWallet.getAccounts();
  }
  signAndExecuteTransaction(transaction) {
    return window.suiWallet.signAndExecuteTransaction(transaction);
  }
  async connect() {
    this.connecting = true;
    if (window.suiWallet) {
      const wallet = window.suiWallet;
      try {
        let given = await wallet.requestPermissions();
        const newLocal = ["viewAccount"];
        let perms = await wallet.hasPermissions(newLocal);
        console.log(perms);
        console.log(given);
        this.connected = true;
      } catch (err) {
        console.error(err);
      } finally {
        this.connecting = false;
      }
    }
  }
  async disconnect() {
    if (this.connected == true) {
      this.connected = false;
    }
    console.log("disconnected");
  }
};
export {
  SuiWalletAdapter
};
